### DISCUSSION TITLE
How to know the ID of a selected sceneitem?

### DISCUSSION THREAD

**Discussion by MarioMey** [upgradeQ/Streaming-Software-Scripting-Reference#26] [Q&A]
Basically, I needed that, so I looked for the code that returns the selected sceneitem as a `obs_sceneitem_t`. In Blender, you can access to objects by their names with `bpy.data.scenes['Scene'].objects['Cube']` but also by the selected and/or active object (you can select some objects and the last selected is the active), by doing `bpy.context.selected_objects[0]`. Is in OBS API something like this? If not...

Another option is to make a signal with a sceneitem... I found it, it is called `item_select`, but I don't know how to use it. There is `obs_get_signal_handler()` for Core OBS Signals and `obs_source_get_signal_handler()` for Cource Signals. Sceneitems signals does exist... but I don't know how to access them. I would make: everytime I select a sceneitem, its ID is printed in console. It is enough for me.

Meanwhile, I'm using `obs_scene_enum_items()` and `obs_sceneitem_get_id()` to get the list of all sceneitems IDs. Also is ok for my need... but I would like to know if other options are possible.

Thanks in advance.

**Comment by upgradeQ** [comment]
There is `obs_sceneitem_selected` which returns boolean state of `obs_sceneitem_t`, though if you hold <kbd>Shift</kbd> and select another source the other one will have "selected" state too.

**Reply by MarioMey** [reply]
Oh, it's a good one! I will try it.
